File: java2csharp/java/Bixolon/UPOS Compliant/BXLTest/CashDrawerFragment.java
package com.bxl.postest;

import jpos.CashDrawer;
import jpos.CashDrawerConst;
import jpos.JposConst;
import jpos.JposException;
import jpos.events.StatusUpdateEvent;
import jpos.events.StatusUpdateListener;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.method.ScrollingMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.TextView;

public class CashDrawerFragment extends Fragment
		implements View.OnClickListener, CompoundButton.OnCheckedChangeListener, StatusUpdateListener {
	
	private CashDrawer cashDrawer;
	
	private EditText logicalNameEditText;
	private TextView stateTextView;
	private TextView deviceMessagesTextView;
	
	private CheckBox deviceEnabledCheckBox;
	
	private CountDownTimer countDownTimer;
	
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		
		cashDrawer = new CashDrawer();
		cashDrawer.addStatusUpdateListener(this);
	}
	
	@Override
	public View onCreateView(LayoutInflater inflater,
			@Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.fragment_cash_drawer, container, false);
		
		SharedPreferences settings = getActivity().getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE);
		logicalNameEditText = (EditText) view.findViewById(R.id.editTextLogicalName);
		logicalNameEditText.setText(settings.getString(MainActivity.KEY_LOGICAL_NAME_POS_PRINTER, getString(R.string.cash_drawer)));
		
		view.findViewById(R.id.buttonOpen).setOnClickListener(this);
		view.findViewById(R.id.buttonClaim).setOnClickListener(this);
		view.findViewById(R.id.buttonRelease).setOnClickListener(this);
		view.findViewById(R.id.buttonClose).setOnClickListener(this);
		view.findViewById(R.id.buttonInfo).setOnClickListener(this);
		view.findViewById(R.id.buttonCheckHealth).setOnClickListener(this);
		
		deviceEnabledCheckBox = (CheckBox) view.findViewById(R.id.checkBoxDeviceEnabled);
		deviceEnabledCheckBox.setOnCheckedChangeListener(this);
		
		view.findViewById(R.id.buttonOpenDrawer).setOnClickListener(this);
		view.findViewById(R.id.buttonGetDrawerOpened).setOnClickListener(this);
		
		stateTextView = (TextView) view.findViewById(R.id.textViewState);
		
		deviceMessagesTextView = (TextView) view.findViewById(R.id.textViewDeviceMessages);
		deviceMessagesTextView.setMovementMethod(new ScrollingMovementMethod());
		deviceMessagesTextView.setVerticalScrollBarEnabled(true);
		return view;
	}
	
	@Override
	public void onDestroyView() {
		super.onDestroyView();
		
		SharedPreferences settings = getActivity().getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE);
		SharedPreferences.Editor editor = settings.edit();
		editor.putString(MainActivity.KEY_LOGICAL_NAME_CASH_DRAWER, logicalNameEditText.getText().toString());
		editor.commit();
	}
	
	@Override
	public void onResume() {
		super.onResume();
		
		countDownTimer = new CountDownTimer(Long.MAX_VALUE, 1000) {
			
			@Override
			public void onTick(long millisUntilFinished) {
				stateTextView.setText(MainActivity.getStatusString(cashDrawer.getState()));
			}
			
			@Override
			public void onFinish() {
				// TODO Auto-generated method stub
				
			}
		}.start();
	}
	
	@Override
	public void onPause() {
		super.onPause();
		
		countDownTimer.cancel();
	}
	
	@Override
	public void onDestroy() {
		super.onDestroy();
		
		try {
			cashDrawer.close();
		} catch (JposException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
	
	@Override
	public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
		try {
			cashDrawer.setDeviceEnabled(isChecked);
		} catch (JposException e) {
			e.printStackTrace();
			try {
				cashDrawer.setDeviceEnabled(!isChecked);
			} catch (JposException e1) {
				// TODO Auto-generated catch block
				e1.printStackTrace();
			}
			MessageDialogFragment.showDialog(getFragmentManager(), "Exception", e.getMessage());
		}
	}

	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.buttonOpen:
			String logicalDeviceName = logicalNameEditText.getText().toString();
			try {
				cashDrawer.open(logicalDeviceName);
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			
			try {
				deviceEnabledCheckBox.setChecked(cashDrawer.getDeviceEnabled());
			} catch (JposException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			break;
			
		case R.id.buttonClaim:
			try {
				cashDrawer.claim(0);
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			break;
			
		case R.id.buttonRelease:
			try {
				cashDrawer.release();
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			break;
			
		case R.id.buttonClose:
			try {
				cashDrawer.close();
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			break;
			
		case R.id.buttonInfo:
			info();
			break;
			
		case R.id.buttonCheckHealth:
			checkHealth();
			break;
			
		case R.id.buttonOpenDrawer:
			try {
				cashDrawer.openDrawer();
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			break;
			
		case R.id.buttonGetDrawerOpened:
			try {
				if (cashDrawer.getDrawerOpened()) {
					deviceMessagesTextView.append("Cash drawer is open.\n");
				} else {
					deviceMessagesTextView.append("Cash drawer is closed.\n");
				}
			} catch (JposException e) {
				e.printStackTrace();
				MessageDialogFragment.showDialog(getFragmentManager(), "Excepction", e.getMessage());
			}
			break;
		}
	}
	
	@Override
	public void statusUpdateOccurred(StatusUpdateEvent e) {
		String text = "Status Update Event: ";
		switch(e.getStatus()){
		case CashDrawerConst.CASH_SUE_DRAWERCLOSED:
			text += "Drawer Closed\n";
			break;
			
		case CashDrawerConst.CASH_SUE_DRAWEROPEN:
			text += "Drawer Opened\n";
			break;
			
		default:
			text += "Unknown Status: " + e.getStatus() + "\n";
			break;
        }
		deviceMessagesTextView.append(text);
	}
	
	private void info() {
		String message;
		try {
			message = "deviceServiceDescription: " + cashDrawer.getDeviceServiceDescription()
					+ "\ndeviceServiceVersion: " + cashDrawer.getDeviceServiceVersion()
					+ "\nphysicalDeviceDescription: " + cashDrawer.getPhysicalDeviceDescription()
					+ "\nphysicalDeviceName: " + cashDrawer.getPhysicalDeviceName()
					+ "\npowerState: " + MainActivity.getPowerStateString(cashDrawer.getPowerState());
			MessageDialogFragment.showDialog(getFragmentManager(), "Info", message);
		} catch (JposException e) {
			e.printStackTrace();
			MessageDialogFragment.showDialog(getFragmentManager(), "Exception",
					"Exception in Info: "+ e.getMessage());
		}
	}
	
	private void checkHealth() {
		try {
			cashDrawer.checkHealth(JposConst.JPOS_CH_INTERNAL);
			MessageDialogFragment.showDialog(getFragmentManager(), "checkHealth",
					cashDrawer.getCheckHealthText());
		} catch (JposException e) {
			e.printStackTrace();
			MessageDialogFragment.showDialog(getFragmentManager(), "Exception", e.getMessage());
		}
	}
}

File: java2csharp/csharp/Bixolon/UPOS Compliant/BXLTest/CashDrawerFragment.cs
namespace com.bxl.postest
{

	using CashDrawer = jpos.CashDrawer;
	using CashDrawerConst = jpos.CashDrawerConst;
	using JposConst = jpos.JposConst;
	using JposException = jpos.JposException;
	using StatusUpdateEvent = jpos.events.StatusUpdateEvent;
	using StatusUpdateListener = jpos.events.StatusUpdateListener;
	using Context = android.content.Context;
	using SharedPreferences = android.content.SharedPreferences;
	using Bundle = android.os.Bundle;
	using CountDownTimer = android.os.CountDownTimer;
	using Nullable = android.support.annotation.Nullable;
	using Fragment = android.support.v4.app.Fragment;
	using ScrollingMovementMethod = android.text.method.ScrollingMovementMethod;
	using LayoutInflater = android.view.LayoutInflater;
	using View = android.view.View;
	using ViewGroup = android.view.ViewGroup;
	using CheckBox = android.widget.CheckBox;
	using CompoundButton = android.widget.CompoundButton;
	using EditText = android.widget.EditText;
	using TextView = android.widget.TextView;

	public class CashDrawerFragment : Fragment, View.OnClickListener, CompoundButton.OnCheckedChangeListener, StatusUpdateListener
	{

		private CashDrawer cashDrawer;

		private EditText logicalNameEditText;
		private TextView stateTextView;
		private TextView deviceMessagesTextView;

		private CheckBox deviceEnabledCheckBox;

		private CountDownTimer countDownTimer;

		public override void onCreate(Bundle savedInstanceState)
		{
			base.onCreate(savedInstanceState);

			cashDrawer = new CashDrawer();
			cashDrawer.addStatusUpdateListener(this);
		}

//JAVA TO C# CONVERTER TODO TASK: Most Java annotations will not have direct .NET equivalent attributes:
//ORIGINAL LINE: @Override public android.view.View onCreateView(android.view.LayoutInflater inflater, @Nullable android.view.ViewGroup container, @Nullable android.os.Bundle savedInstanceState)
		public override View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			View view = inflater.inflate(R.layout.fragment_cash_drawer, container, false);

			SharedPreferences settings = Activity.getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE);
			logicalNameEditText = (EditText) view.findViewById(R.id.editTextLogicalName);
			logicalNameEditText.Text = settings.getString(MainActivity.KEY_LOGICAL_NAME_POS_PRINTER, getString(R.@string.cash_drawer));

			view.findViewById(R.id.buttonOpen).OnClickListener = this;
			view.findViewById(R.id.buttonClaim).OnClickListener = this;
			view.findViewById(R.id.buttonRelease).OnClickListener = this;
			view.findViewById(R.id.buttonClose).OnClickListener = this;
			view.findViewById(R.id.buttonInfo).OnClickListener = this;
			view.findViewById(R.id.buttonCheckHealth).OnClickListener = this;

			deviceEnabledCheckBox = (CheckBox) view.findViewById(R.id.checkBoxDeviceEnabled);
			deviceEnabledCheckBox.OnCheckedChangeListener = this;

			view.findViewById(R.id.buttonOpenDrawer).OnClickListener = this;
			view.findViewById(R.id.buttonGetDrawerOpened).OnClickListener = this;

			stateTextView = (TextView) view.findViewById(R.id.textViewState);

			deviceMessagesTextView = (TextView) view.findViewById(R.id.textViewDeviceMessages);
			deviceMessagesTextView.MovementMethod = new ScrollingMovementMethod();
			deviceMessagesTextView.VerticalScrollBarEnabled = true;
			return view;
		}

		public override void onDestroyView()
		{
			base.onDestroyView();

			SharedPreferences settings = Activity.getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE);
			SharedPreferences.Editor editor = settings.edit();
			editor.putString(MainActivity.KEY_LOGICAL_NAME_CASH_DRAWER, logicalNameEditText.Text.ToString());
			editor.commit();
		}

		public override void onResume()
		{
			base.onResume();

			countDownTimer = new CountDownTimerAnonymousInnerClassHelper(this, long.MaxValue)
			.start();
		}

		private class CountDownTimerAnonymousInnerClassHelper : CountDownTimer
		{
			private readonly CashDrawerFragment outerInstance;

			public CountDownTimerAnonymousInnerClassHelper(CashDrawerFragment outerInstance, UnknownType MAX_VALUE) : base(MAX_VALUE, 1000)
			{
				this.outerInstance = outerInstance;
			}


			public override void onTick(long millisUntilFinished)
			{
				outerInstance.stateTextView.Text = MainActivity.getStatusString(outerInstance.cashDrawer.State);
			}

			public override void onFinish()
			{
				// TODO Auto-generated method stub

			}
		}

		public override void onPause()
		{
			base.onPause();

			countDownTimer.cancel();
		}

		public override void onDestroy()
		{
			base.onDestroy();

			try
			{
				cashDrawer.close();
			}
			catch (JposException e)
			{
				// TODO Auto-generated catch block
				Console.WriteLine(e.ToString());
				Console.Write(e.StackTrace);
			}
		}

		public override void onCheckedChanged(CompoundButton buttonView, bool isChecked)
		{
			try
			{
				cashDrawer.DeviceEnabled = isChecked;
			}
			catch (JposException e)
			{
				Console.WriteLine(e.ToString());
				Console.Write(e.StackTrace);
				try
				{
					cashDrawer.DeviceEnabled = !isChecked;
				}
				catch (JposException e1)
				{
					// TODO Auto-generated catch block
					Console.WriteLine(e1.ToString());
					Console.Write(e1.StackTrace);
				}
				MessageDialogFragment.showDialog(FragmentManager, "Exception", e.Message);
			}
		}

		public override void onClick(View v)
		{
			switch (v.Id)
			{
			case R.id.buttonOpen:
				string logicalDeviceName = logicalNameEditText.Text.ToString();
				try
				{
					cashDrawer.open(logicalDeviceName);
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}

				try
				{
					deviceEnabledCheckBox.Checked = cashDrawer.DeviceEnabled;
				}
				catch (JposException e)
				{
					// TODO Auto-generated catch block
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
				}
				break;

			case R.id.buttonClaim:
				try
				{
					cashDrawer.claim(0);
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}
				break;

			case R.id.buttonRelease:
				try
				{
					cashDrawer.release();
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}
				break;

			case R.id.buttonClose:
				try
				{
					cashDrawer.close();
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}
				break;

			case R.id.buttonInfo:
				info();
				break;

			case R.id.buttonCheckHealth:
				checkHealth();
				break;

			case R.id.buttonOpenDrawer:
				try
				{
					cashDrawer.openDrawer();
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}
				break;

			case R.id.buttonGetDrawerOpened:
				try
				{
					if (cashDrawer.DrawerOpened)
					{
						deviceMessagesTextView.append("Cash drawer is open.\n");
					}
					else
					{
						deviceMessagesTextView.append("Cash drawer is closed.\n");
					}
				}
				catch (JposException e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
					MessageDialogFragment.showDialog(FragmentManager, "Excepction", e.Message);
				}
				break;
			}
		}

		public override void statusUpdateOccurred(StatusUpdateEvent e)
		{
			string text = "Status Update Event: ";
			switch (e.Status)
			{
			case CashDrawerConst.CASH_SUE_DRAWERCLOSED:
				text += "Drawer Closed\n";
				break;

			case CashDrawerConst.CASH_SUE_DRAWEROPEN:
				text += "Drawer Opened\n";
				break;

			default:
				text += "Unknown Status: " + e.Status + "\n";
				break;
			}
			deviceMessagesTextView.append(text);
		}

		private void info()
		{
			string message;
			try
			{
				message = "deviceServiceDescription: " + cashDrawer.DeviceServiceDescription + "\ndeviceServiceVersion: " + cashDrawer.DeviceServiceVersion + "\nphysicalDeviceDescription: " + cashDrawer.PhysicalDeviceDescription + "\nphysicalDeviceName: " + cashDrawer.PhysicalDeviceName + "\npowerState: " + MainActivity.getPowerStateString(cashDrawer.PowerState);
				MessageDialogFragment.showDialog(FragmentManager, "Info", message);
			}
			catch (JposException e)
			{
				Console.WriteLine(e.ToString());
				Console.Write(e.StackTrace);
				MessageDialogFragment.showDialog(FragmentManager, "Exception", "Exception in Info: " + e.Message);
			}
		}

		private void checkHealth()
		{
			try
			{
				cashDrawer.checkHealth(JposConst.JPOS_CH_INTERNAL);
				MessageDialogFragment.showDialog(FragmentManager, "checkHealth", cashDrawer.CheckHealthText);
			}
			catch (JposException e)
			{
				Console.WriteLine(e.ToString());
				Console.Write(e.StackTrace);
				MessageDialogFragment.showDialog(FragmentManager, "Exception", e.Message);
			}
		}
	}

}